### PR TITLE
Port a few testcases, fix ==> in multiple assertions

### PR DESCRIPTION
**Pull request by karlosss**

### CODE REVIEW

File: prusti-specs/src/specifications/preparser.rs [hunk 1]
```diff
@@ -119,9 +119,9 @@ impl ParserStream {
                 if punct.as_char() != c {
                     return false;
                 }
-                if i + 1 < operator.len() && punct.spacing() != Spacing::Joint {
-                    return false;
-                }
+                // if i + 1 < operator.len() && punct.spacing() != Spacing::Joint {
+                //     return false;
+                // }
```
Comment (Aurel300) [inline prusti-specs/src/specifications/preparser.rs:130]: Better to delete or add a comment explaining why the code is disabled.

File: prusti-viper/src/encoder/encoder.rs [hunk 2]
```diff
@@ -652,47 +652,47 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
...
+                final_function.unwrap()
+            });
+        vir::Expr::FuncApp(
+            name,
+            vec![place],
+            vec![self_local_var],
+            vir::Type::Int,
+            vir::Position::default(),
+        )
+        // unimplemented!();
```
Comment (Aurel300) [inline prusti-viper/src/encoder/encoder.rs]: Can just be deleted.

Comment (karlosss) [reply prusti-viper/src/encoder/encoder.rs]: Thanks, forgot this.

File: prusti-specs/src/specifications/preparser.rs (hunk 1 above)
Comment (karlosss) [reply prusti-specs/src/specifications/preparser.rs:130]: Commented on that one.

File: prusti-viper/src/encoder/type_encoder.rs [hunk 3]
```diff
@@ -701,28 +701,28 @@ impl<'p, 'v, 'r: 'v, 'tcx: 'v> TypeEncoder<'p, 'v, 'tcx> {
...
+            discr_values.push(signed_discr);
+        }
+    } else {
+        for (variant_index, _) in adt_def.variants.iter().enumerate() {
+            let value = adt_def.discriminant_for_variant(tcx, abi::VariantIdx::from_usize(variant_index)).val;
+            discr_values.push(value as i128);
+        }
+    }
+    discr_values
+    // unimplemented!();
```
Comment (Aurel300) [inline prusti-viper/src/encoder/type_encoder.rs]: And here :)

**Review by vakaras — APPROVED** [review]

**Pull request merged**